distributed/ColdStartDistributedTraining.ipynb

```diff
@@ -0,0 +1,320 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# [Distributed] Pre-training and fine-tuning an LLM on CPU on AG News with ThirdAI's UDT\n",
+    "\n",
+    "This training script shows how to run data parallel pre-training of an LLM from scratch on the popular AG News Dataset (https://www.kaggle.com/datasets/amananandrai/ag-news-classification-dataset) using ThirdAI's Universal Deep Transformer (UDT)."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. Import thirdai and activate license"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip3 install datasets\n",
+    "# !pip3 install thirdai --upgrade\n",
+    "!pip3 install ray\n",
+    "\n",
+    "import thirdai\n",
+    "# activation key here"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2. Ray Cluster Initialization\n",
+    "### For the purpose of this demo, we will be initializing a mock ray cluster here. <br>\n",
+    "### Change num_cpus accordingly."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from ray.cluster_utils import Cluster\n",
+    "\n",
+    "mini_cluster = Cluster(\n",
+    "    initialize_head=True,\n",
+    "    head_node_args={\n",
+    "        \"num_cpus\": 24,\n",
+    "    },\n",
+    ")\n",
+    "mini_cluster.add_node(num_cpus=24)"
```

Comment:
We might be also running these on the CI too, make sure to decrease the number of CPUs to 2.

Reply:
Done.